cmd/src/batch_common.go

```diff
@@ -292,29 +292,45 @@ func executeBatchSpec(ctx context.Context, ui ui.ExecUI, opts executeBatchSpecOp
 
 	// EXECUTION OF TASKS
 	coord := svc.NewCoordinator(executor.NewCoordinatorOpts{
-		Creator:          workspaceCreator,
-		CacheDir:         opts.flags.cacheDir,
-		Cache:            executor.NewDiskCache(opts.flags.cacheDir),
-		ClearCache:       opts.flags.clearCache,
-		SkipErrors:       opts.flags.skipErrors,
-		CleanArchives:    opts.flags.cleanArchives,
-		Parallelism:      opts.flags.parallelism,
-		Timeout:          opts.flags.timeout,
-		KeepLogs:         opts.flags.keepLogs,
-		TempDir:          opts.flags.tempDir,
-		ImportChangesets: true,
+		Creator:       workspaceCreator,
+		CacheDir:      opts.flags.cacheDir,
+		Cache:         executor.NewDiskCache(opts.flags.cacheDir),
+		SkipErrors:    opts.flags.skipErrors,
+		CleanArchives: opts.flags.cleanArchives,
+		Parallelism:   opts.flags.parallelism,
+		Timeout:       opts.flags.timeout,
+		KeepLogs:      opts.flags.keepLogs,
+		TempDir:       opts.flags.tempDir,
 	})
 
 	ui.CheckingCache()
 	tasks := svc.BuildTasks(ctx, batchSpec, workspaces)
-	uncachedTasks, cachedSpecs, err := coord.CheckCache(ctx, tasks)
-	if err != nil {
-		return err
+	var (
+		specs         []*batcheslib.ChangesetSpec
+		uncachedTasks []*executor.Task
+	)
+	if opts.flags.clearCache {
+		coord.ClearCache(ctx, tasks)
+		uncachedTasks = tasks
+	} else {
+		uncachedTasks, specs, err = coord.CheckCache(ctx, tasks)
+		if err != nil {
+			return err
+		}
 	}
-	ui.CheckingCacheSuccess(len(cachedSpecs), len(uncachedTasks))
+	ui.CheckingCacheSuccess(len(specs), len(uncachedTasks))
 
 	taskExecUI := ui.ExecutingTasks(*verbose, opts.flags.parallelism)
-	freshSpecs, logFiles, err := coord.Execute(ctx, uncachedTasks, batchSpec, taskExecUI)
+	freshSpecs, logFiles, execErr := coord.Execute(ctx, uncachedTasks, batchSpec, taskExecUI)
+	// Add external changeset specs.
+	importedSpecs, importErr := svc.CreateImportChangesetSpecs(ctx, batchSpec)
+	err = nil
+	if execErr != nil {
+		err = multierror.Append(err, execErr)
+	}
+	if importErr != nil {
+		err = multierror.Append(err, importErr)
+	}
 	if err != nil && !opts.flags.skipErrors {
```

Comment:
Does this still work or would you need to do a `err.ErrorOrNil()` after declaring err to be a multierror?

There's also some error-unpacking going on the TUI to pretty-print errors. Does this work with mulitierror?

Reply:
Fixed this!